evm_arithmetization/src/witness/operation.rs

```diff
@@ -1009,32 +855,37 @@ pub(crate) fn generate_mstore_general<F: Field>(
     Ok(())
 }
 
-pub(crate) fn generate_mstore_32bytes<F: Field>(
+pub(crate) fn generate_mstore_32bytes<F: Field, S: State<F>>(
     n: u8,
-    state: &mut GenerationState<F>,
+    state: &mut S,
     mut row: CpuColumnsView<F>,
 ) -> Result<(), ProgramError> {
-    let [(addr, _), (val, log_in1)] = stack_pop_with_log_and_fill::<2, _>(state, &mut row)?;
+    let generation_state = state.get_mut_generation_state();
+    let [(addr, _), (val, log_in1)] =
+        stack_pop_with_log_and_fill::<2, _>(generation_state, &mut row)?;
 
     let base_address = MemoryAddress::new_bundle(addr)?;
 
-    byte_unpacking_log(state, base_address, val, n as usize);
+    byte_unpacking_log(generation_state, base_address, val, n as usize);
 
     let new_addr = addr + n;
-    push_no_write(state, new_addr);
+    push_no_write(generation_state, new_addr);
 
-    state.traces.push_memory(log_in1);
-    state.traces.push_cpu(row);
+    generation_state.traces.push_memory(log_in1);
+    generation_state.traces.push_cpu(row);
     Ok(())
 }
 
-pub(crate) fn generate_exception<F: Field>(
+pub(crate) fn generate_exception<F: Field, T: Transition<F>>(
     exc_code: u8,
-    state: &mut GenerationState<F>,
+    state: &mut T,
     mut row: CpuColumnsView<F>,
     is_generation: bool,
 ) -> Result<(), ProgramError> {
-    if TryInto::<u32>::try_into(state.registers.gas_used).is_err() {
+    // TDO: se fue arriba
```

Comment:
I was missing this TODO! which is actually a question for @linda.
I had to move the call to fill_stack_fiels because of "borrowing mutable twice" errors. Do you if it's safe to do this?

Reply:
It should be safe yes: fill_stack_fields only adds memory operations related to check_overflow and is_stack_top_read when necessary.